[Rithik Lenin] iP

rithiklenin opened the pull request:
### Optimus

> _"The supreme art of war is to subdue the enemy without fighting."_ - Sun Tzu

Optimus allows you to do anything you wish. It's,

1. text-based
2. easy to learn
3. FAST SUPER FAST to use
4. All you need to do is,

Download it from [here](https://github.com/rithiklenin/ip/releases/tag/v0.1).
AND let it manage your tasks for you 😉
WHATS MORE..... it is FREE!

Features:

 

- [ ] Managing tasks
- [ ]  Managing deadlines (Upcoming feature)
- [ ]  Reminders (Upcoming feature)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public static void main(String[] args) {
        new Optimus(FILE_PATH).run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

rithiklenin committed:
changed filename to Optimus.java

rithiklenin committed:
changed text in file to print required things in Optimus.java

rithiklenin committed:
added the next level features for file

rithiklenin committed:
added changes to file to have added functionality but having some problem with list

rithiklenin committed:
Added additional function but there still seems to be some bug

rithiklenin committed:
Added changes and completed level 2

rithiklenin committed:
Changed some functionalities for lvl 3

rithiklenin committed:
Added 3 classes to help with array and task management

rithiklenin committed:
Removed Record file and added for the rest

rithiklenin committed:
switched bye

rithiklenin committed:
added mark and unmark

rithiklenin committed:
Added the final logic for level 3

rithiklenin committed:
made edits to add new classes

rithiklenin committed:
added some functions which override for all classes

rithiklenin committed:
Added all the 3 classes and made them all work

rithiklenin committed:
testing complete

rithiklenin committed:
added exception for mark and unmark is number not in list

rithiklenin committed:
added exceptions to other parts of my file to handle 5 different errors

rithiklenin committed:
Changed array to arrayList in Main class

rithiklenin committed:
Added delete functionality in Main file

rithiklenin committed:
Add file reading function

rithiklenin committed:
Optimus.java: Add separate method loadFile to handle file loading

rithiklenin committed:
Task.java: Add new String method to save to file

rithiklenin committed:
Optimus.java: Remove count from load function and move to main function. Add new save function

rithiklenin committed:
Task.java: Add toSaveString function to print file format
Deadlines.java: Add toSaveString from abstract class Task.java
Events.java: Add toSaveString from abstract class Task.java
ToDos.java: Add toSaveString from abstract class Task.java

rithiklenin committed:
optimus.txt: Save tasks from user onto file

The user inputs were stored in the text file which can later be retrieved to be used.

rithiklenin committed:
Deadline.java: Update printed string
Events.java: Update printed string

Fixed the format to be saved after the dates and time were found not to be saved.

rithiklenin committed:
Optimus.java: Level-7 Save

Completed the load and save needed for level 7.

rithiklenin committed:
Optimus.java: Add comments for lines

rithiklenin committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Optimus.java: Add comments for lines
  Optimus.java: Level-7 Save
  Deadline.java: Update printed string Events.java: Update printed string
  optimus.txt: Save tasks from user onto file
  Task.java: Add toSaveString function to print file format Deadlines.java: Add toSaveString from abstract class Task.java Events.java: Add toSaveString from abstract class Task.java ToDos.java: Add toSaveString from abstract class Task.java
  Optimus.java: Remove count from load function and move to main function. Add new save function
  Task.java: Add new String method to save to file
  Optimus.java: Add separate method loadFile to handle file loading
  Add file reading function

rithiklenin committed:
optimus.txt: Change file after execution
Deadlines.java: Add imports for Level 8 from notes
Optimus.java: Change parts of code to reflect what is needed

There was no additional functionalities. They are minor changes

rithiklenin committed:
Merge branch 'master' of https://github.com/rithiklenin/ip into branch-Level-8

* 'master' of https://github.com/rithiklenin/ip:
  Optimus.java: Add comments for lines

# Conflicts:
#	src/main/java/Optimus.java

Conflicts have been resolved

rithiklenin committed:
optimus.txt: Change details in file from program running
Deadlines.java: Add feature to convert string to time
Events.java: Add feature to convert string to time
Optimus java: Add feature to accommodate feature changes in Deadlines and Events

The files were changed to reflect the Level 8 changes.

rithiklenin committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  optimus.txt: Change details in file from program running Deadlines.java: Add feature to convert string to time Events.java: Add feature to convert string to time Optimus java: Add feature to accommodate feature changes in Deadlines and Events
  optimus.txt: Change file after execution Deadlines.java: Add imports for Level 8 from notes Optimus.java: Change parts of code to reflect what is needed

rithiklenin committed:
optimus.txt: Change file due to running program
Optimus.java: Reduce lines due to added classes
Parser.java: Handle main logic of program
Storage.java: Handle load and save
TaskList.java: Add commands related to record list
Ui.java: Add strings and exceptions

The following files were edited to be able to refactor the code. The changes now make it simpler to read the code in main Optimus.java file.

rithiklenin committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  optimus.txt: Change file due to running program Optimus.java: Reduce lines due to added classes Parser.java: Handle main logic of program Storage.java: Handle load and save TaskList.java: Add commands related to record list Ui.java: Add strings and exceptions

rithiklenin committed:
Move to optimus package

The following was done for A-Packages

rithiklenin committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Move to optimus package

rithiklenin committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

rithiklenin committed:
build.grade: Help support automated testing
optimus.txt: Add changes after program
Parser.java: Change logic for Event
DeadlineTest.java: Add JUnit to test method
EventsTest.java: Add JUnit to test method

Completed JUnit testing for 2 methods using the following files above.

rithiklenin committed:
Merge branch 'branch-A-JUnit'

* branch-A-JUnit:
  build.grade: Help support automated testing optimus.txt: Add changes after program Parser.java: Change logic for Event DeadlineTest.java: Add JUnit to test method EventsTest.java: Add JUnit to test method

rithiklenin committed:
Add JavaDocs for all files

The following commit files were edited to add the javadoc strings into the respective files.

rithiklenin committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDocs for all files

rithiklenin committed:
All files: Change Optimus to optimus to obey coding standard for package

The following files were all edited to comply with the coding standards.

rithiklenin committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  All files: Change Optimus to optimus to obey coding standard for package

rithiklenin committed:
Parser.java: Add word "find" to search for tasks with similar keywords
TaskList.java: Add logic to find tasks
Ui.java: Add messages to print to user

The following classes were edited to ensure that the user is able to receive their intended search keyword tasks. This completes the level 9 changes. The methods added were printTasks in Ui, findTasks in TaskList.

rithiklenin committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Parser.java: Add word "find" to search for tasks with similar keywords TaskList.java: Add logic to find tasks Ui.java: Add messages to print to user

rithiklenin committed:
All files: Change structure

Minor refactoring of code.

rithiklenin committed:
All Files: Refactor code in some areas

The following files were edited to adjust some indentation errors.

rithiklenin committed:
build.gradle: Add checkstyle plugin

Added the changes in accordance to the website for checkstyle using gradle.

rithiklenin committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  build.gradle: Add checkstyle plugin

Sornsornah commented on src/Test/java/optimus/DeadlinesTest.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Test
+    public void testParseStringValidFormat1() throws OptimusException {
```
Should the test method name follow the format of `featureUnderTest_testScenario_expectedBehavior()`?

Sornsornah commented on src/main/java/optimus/Deadlines.java:
```diff
@@ -0,0 +1,67 @@
+     * Returns the string representation of the deadline task formatted for saving to a file.
+     */
+    @Override
+    public String toSaveString() {
```
Should the method name sound more like a getter method, e.g. `getFileString()`?

Sornsornah commented on src/main/java/optimus/Parser.java:
```diff
@@ -0,0 +1,104 @@
+            }
+            record.findTasks(keyword, ui);
+        } else if (text.equals("list")) {
+            ui.listTasks(record);
```
I like how the code is quite modular and broken down into the different functions! 👍

Sornsornah commented on src/main/java/optimus/Parser.java:
```diff
@@ -0,0 +1,104 @@
+        } else if (text.startsWith("event ")) {
+            String[] parts = text.split(" /from | /to ");
+            if (parts.length < 3) {
+                throw new OptimusException("The event time format is incorrect. " +
+                        "Please provide both 'from' and 'to' times (e.g., event meeting /from 2024-  " +
+                        "2024-09-10 14:00 /to 2024-09-10 16:00).");
```
I like how your error messages are very detailed and gives the user the correct format for inputs! 😄

Sornsornah commented on src/main/java/optimus/TaskList.java:
```diff
@@ -0,0 +1,50 @@
+        return record.get(index);
+    }
+
+    public int sizeOfRecord() {
```
Should the method name be modified to sound like a getter (e.g. getSizeOfRecord())?

Sornsornah commented on src/main/java/optimus/ToDos.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * Represents a to-do task in the Optimus application.
+ */
+public class ToDos extends Task {
```
Should the class name be singular?

Sornsornah commented on src/main/java/optimus/Ui.java:
```diff
@@ -0,0 +1,121 @@
+     * @param task the task that has been added.
+     * @param size the current number of tasks in the list.
+     */
+    public void TaskAdded(Task task, int size) {
```
Should method name sound more like an action (e.g. `displayTaskAdded()`)?

Sornsornah reviewed (commented):
Overall, I found your code quite easy to understand as it was broken down into the different functions. I noted a few naming conventions that could be improved on.

Justincjr commented on src/main/java/optimus/Ui.java:
class methods should follow camelCase

Justincjr commented on src/main/java/optimus/Ui.java:
```diff
@@ -0,0 +1,121 @@
+     */
+    public void listFoundTasks(List<Task> tasks) {
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
```
should use the getSize class method instead

Justincjr commented on src/main/java/optimus/Task.java:
```diff
@@ -0,0 +1,67 @@
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
Maybe can name it isMark instead to be clearer

Justincjr commented on src/Test/java/optimus/DeadlinesTest.java:
can add test for invalid formats that should throw OptimusException

Justincjr commented on src/Test/java/optimus/EventsTest.java:
can add test for invalid formats that should throw OptimusException

rithiklenin committed:
All files: Add relevant functions to build GUI

The committed files were all edited to build a function GUI.

rithiklenin committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  All files: Add relevant functions to build GUI

rithiklenin committed:
Optimus.java: Add assert statements for UI and Storage
Parser.java: Add assert statements for text

Added assert statements in `Optimus.java` for UI and Storage initialisation. These ensure that both the UI and Storage components are correctly initialised before proceeding with task management operations. This guards against null references and ensures system stability during runtime.

Introduced assert statements in `Parser.java` to verify that the text input passed for command parsing is not null or empty. This adds an additional layer of validation, ensuring that the Parser receives valid input, preventing potential command handling errors down the line.

These assertions improve robustness and catch potential errors early during execution, providing clearer debugging information when encountered.

rithiklenin committed:
Optimus.txt: Change file based on user input
Parser.java: Add refactoring to breakdown code

The files were changed to reduce the total number of lines. The Parser
class was broken down to many methods that deal directly with the main
functionality of the code.

rithiklenin committed:
Merge pull request #1 from rithiklenin/branch-A-Assertions

Optimus.java: Add assert statements for UI and Storage

rithiklenin committed:
Merge pull request #2 from rithiklenin/branch-A-CodeQuality

Optimus.txt: Change file based on user input

rithiklenin committed:
DialogBox.java: Change variable in class to allow function to work

rithiklenin committed:
TaskList.java: Add functionality to improve search

In the following class, search was improved so that it can now detect
a series of words to better match to the tasks. Hence, it will be much
narrower than before and gives better results.

rithiklenin committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  TaskList.java: Add functionality to improve search

rithiklenin committed:
All files: Add changes to functionalities by editing description error
throwing.
The exception thrown was giving the wrong output which has now been
rectified.

rithiklenin committed:
Storage.java: Refactor code
The storage class was edited to reduce number of lines in each method.

rithiklenin committed:
Parser.java: Add functionality to detect no text entered
TaskListTest.java: Add Junit tests

rithiklenin committed:
All files: Add documentation to help users understand functions

rithiklenin committed:
Merge branch 'branch-A-AiAssisted'

* branch-A-AiAssisted:
  All files: Add documentation to help users understand functions

rithiklenin committed:
All files: Add additional functionalities to improve on previous
code

rithiklenin committed:
All files: Add changes to README to reflect on the github page

rithiklenin committed:
README.md: Change features to remove lines after headings

rithiklenin committed:
All files: Update functionalities to improve the performance of code

rithiklenin committed:
All files: Add documentation to improve user readability

rithiklenin committed:
Optimus.java: Add main method

rithiklenin committed:
build.gradle: Change main class to be launcher

rithiklenin committed:
All files: Add new functionality to handle storage which was throwing
an error

rithiklenin committed:
All files: Add new code to improve functionality

rithiklenin committed:
All files: Remove additional spaces

rithiklenin committed:
All files: Change files to correct code style

rithiklenin committed:
All files: Add functionality to deal with number exception error

rithiklenin committed:
README.md: Change to different introduction

rithiklenin committed:
All files: Change comments to reflect code style

rithiklenin committed:
All files: Move import statements up

rithiklenin committed:
TaskListTest.java: Change parts of code to reflect style

rithiklenin committed:
All files: Change code style in JUnit files

rithiklenin committed:
README.md: Change file to suit my bot

rithiklenin committed:
README.md: Remove picture to fit style
